config/pmd.xml

```diff
@@ -320,22 +320,27 @@
     <properties>
       <property name="maxmethods" value="20"/>
       <!-- The class PackageObjectFactory has many boilerplate methods.
-           JavadocMethodCheck is in the process of being rewritten. -->
+           JavadocMethodCheck is in the process of being rewritten.
+           UnusedLocalVariable has some methods that would be moved to util
```

Comment:
Please share link to discussion about this.

Reply:
There was no major discussion, https://github.com/checkstyle/checkstyle/pull/10930#issuecomment-1000910938 and some following comments, the methods that will be moved to util have it written in their Javadoc, example is mentioned below

Reply:
Thanks for explaining. Let's make an issue for this now so that we do not forget. If you think that we can remove suppression for `ExcessiveClassLength` once methods are moved to util class, let's mention that in the issue as well. Then we can change the comments in this file to simple `until...`

Reply:
Also include the nested static class ClassDesc, final class check also contains a class called ClassDesc, we may merge both of these classes and extract it somewhere, there are some overlapping properties. I have also mentioned this in Javadoc of the class.

Reply:
@nmancus1 Created issue at  #11201

Reply:
@nmancus1 The confusion that whether we would be able to remove `MethodCount` suppression is due to this conflicting rule-
https://github.com/checkstyle/checkstyle/blob/fc65adcd5630e12ead59561abceb6794211c95f0/config/checkstyle_checks.xml#L794-L795

https://github.com/checkstyle/checkstyle/blob/fc65adcd5630e12ead59561abceb6794211c95f0/config/pmd.xml#L319-L321

Checkstyle has a limit of 34 and PMD has a limit of 20, we would be able to get under 34 but not under 20.

Reply:
Thanks for explaining, is there some nuance of the pmd check that requires different maximum here?

Reply:
I don't think so, I would have accepted if it was 34 for PMD and 20 for CS as https://checkstyle.sourceforge.io/config_sizes.html#MethodCount calculated method count for only main type declaration parent, if there is a nested class and that has a method, it won't increase the count for the outer class whereas PMD on the other hand just calculated the total number of methods
https://pmd.github.io/latest/pmd_rules_java_design.html#toomanymethods (see the XPath Expr part).
I feel a new issue should be raised regarding this.